state-sync/state-sync-v2/data-streaming-service/src/data_stream.rs

```diff
@@ -109,11 +109,11 @@ impl<T: AptosDataClient + Send + Clone + 'static> DataStream<T> {
     ) -> Result<(Self, DataStreamListener), Error> {
         // Create a new data stream listener
         let (notification_sender, notification_receiver) = aptos_channel::new(
-            QueueStyle::KLAST,
+            QueueStyle::FIFO, // If the stream overflows, drop the new messages
```

Comment:
probably should just close the stream when overflows? the gap is not recoverable anyway

Reply:
Yeah, it'll automatically happen when the consumer (i.e., the driver) fetches all the data and then hits the non-contiguous part, where it'll terminate the stream.